fedimint-client/src/db.rs

```diff
@@ -432,7 +432,15 @@ pub async fn apply_migrations_core_client(
     target_version: DatabaseVersion,
     migrations: BTreeMap<DatabaseVersion, CoreMigrationFn>,
 ) -> Result<(), anyhow::Error> {
-    apply_migrations(db, kind, target_version, migrations, None).await
+    apply_migrations(
+        db,
+        kind,
+        target_version,
+        migrations,
+        None,
+        Some(DbKeyPrefix::UserData as u8),
```

Comment:
This will also filter out `InternalReservedStart` and `ModuleGlobalPrefix`. Will that cause any issues for detecting if it's a new db?